pkg/common/log/hclog_adapter.go

```diff
@@ -66,13 +89,28 @@ func (a *HCLogAdapter) SetLevel(hclog.Level) {
 
 func (a *HCLogAdapter) With(args ...interface{}) hclog.Logger {
 	e := a.CreateEntry(args)
-	return &HCLogAdapter{Log: e}
+	newArgs := make([]interface{}, len(a.args)+len(args))
+	copy(newArgs, a.args)
+	copy(newArgs[len(a.args):], args)
+	return &HCLogAdapter{
+		log:  e,
+		args: newArgs,
+	}
```

Comment:
Thanks for this, @yurishkuro.

Reply:
Reusing the array is precisely what I was trying to avoid, because a given logger might be sub-scoped multiple times via `With()`, and each of the sub-loggers should have its own unique `args`, not override each other.

Case in point of append override: https://play.golang.org/p/z3dKDE9wQcF

Reply:
:facepalm: yep, good call.